joins.qmd

```diff
@@ -606,78 +607,26 @@ There are three possible outcomes for a row in `x`:
 -   If it matches 1 row in `y`, it's preserved.
 -   If it matches more than 1 row in `y`, it's duplicated once for each match.
 
-In principle, this means that there's no guaranteed correspondence between the rows in the output and the rows in the `x`:
-
--   There might be fewer rows if some rows in `x` don't match any rows in `y`.
--   There might be more rows if some rows in `x` match multiple rows in `y`.
--   There might be the same number of rows if every row in `x` matches one row in `y`.
--   There might be the same number of rows if some rows don't match any rows, and exactly the same number of rows match two rows in `y`!!
-
-Row expansion is a fundamental property of joins, but it's dangerous because it might happen without you realizing it.
-To avoid this problem, dplyr will warn whenever there are multiple matches:
+In principle, this means that there's no guaranteed correspondence between the rows in the output and the rows in `x`, but in practice, this rarely causes problems.
+There is, however, one particularly dangerous case which can cause a combinatorial explosion of rows.
+Imagine joining the following two tables:
 
 ```{r}
-df1 <- tibble(key = c(1, 2, 3), val_x = c("x1", "x2", "x3"))
+df1 <- tibble(key = c(1, 2, 2), val_x = c("x1", "x2", "x3"))
 df2 <- tibble(key = c(1, 2, 2), val_y = c("y1", "y2", "y3"))
-
-df1 |> 
-  inner_join(df2, join_by(key))
 ```
 
-You can gain further control over row matching with two arguments:
-
--   `unmatched` controls what happens when a row in `x` fails to match any rows in `y`. It defaults to `"drop"` which will silently drop any unmatched rows.
--   `multiple` controls what happens when a row in `x` matches more than one row in `y`. For equi-joins, it defaults to `"warn"` which emits a warning message if any rows have multiple matches.
-
-There are two common cases in which you might want to override these defaults: enforcing a one-to-one mapping or deliberately allowing the rows to increase.
-
-### One-to-one mapping
-
-Both `unmatched` and `multiple` can take value `"error"` which means that the join will fail unless each row in `x` matches exactly one row in `y`:
+While the first row in `df1` only matches one row in `df2`, the second and third rows both match two rows.
+This is sometimes called a `many-to-many` join, and will cause dplyr to emit a warning:
 
 ```{r}
-#| error: true
-df1 <- tibble(x = 1)
-df2 <- tibble(x = c(1, 1))
-df3 <- tibble(x = 3)
-
 df1 |> 
-  inner_join(df2, join_by(x), unmatched = "error", multiple = "error")
-df1 |> 
-  inner_join(df3, join_by(x), unmatched = "error", multiple = "error")
-```
-
-Note that `unmatched = "error"` is not useful with `left_join()` because, as described above, every row in `x` has a fallback match to a virtual row in `y`.
```

Comment:
Not exactly true anyways, for `left_join()` it will error if a row in `y` was about to be dropped

It isn't useful for `full_join()` (all rows are always kept), so it isn't an argument to that join function.